module/VuFind/src/VuFind/Sitemap/Plugin/Index/CursorMarkIdFetcher.php

```diff
@@ -141,7 +141,7 @@ public function getIdsFromBackend(
         foreach ($filters as $filter) {
             $params->add('fq', $filter);
         }
-        $command = new GetIdsCommand(
+        $command = new GetSitemapFieldsCommand(
```

Comment:
Just a thought, but what if we made the existing GetIdsCommand (and associated backend logic) just a little bit smarter, so if we pass a 'fl' value in through the extra ParamBag, it uses that (maybe with logic to prepend the unique ID field if that's not already found in the passed value). Then instead of having to build all the infrastructure for GetSitemapFieldsCommand, we could just set an extra field in the ParamBag we send to GetIdsCommand.